### ISSUE TITLE
Access Role module no longer support "Any" for the machines field

### ISSUE THREAD

**Issue by alshawwaf** [CheckPointSW/CheckPointAnsibleMgmtCollection#79]
Prior to the update from June, we were able to use "any" for the machines as below

    - name: set access roles
      check_point.mgmt.cp_mgmt_access_role:
        name: "FWadmins"
        color: "blue"
        comments: "Test via Ansible"
        machines: "any"
        tags: access_roles

However, since the last update, it seems that we are making it a requirement to receive a list of objects and the playbook will fail with the following error

"msg": "dictionary requested, could not parse JSON or key=value"

Thanks!
Khalid

**Comment by chkp-edenbr** [comment]
Hi Khalid,

Thanks for updating us we will investigate this.

Regards,
Eden

**Comment by arw-thomasm** [comment]
Hi,

you are requesting a list of dictionaries for machines and users in the module specs. That's obviously the reason, why a string like "any" or "all identified" do not work:

From `cp_mgmt_access_role.py`:

```
def main():
    argument_spec = dict(

        [...]

        machines=dict(type='list', elements='dict', options=dict(
            source=dict(type='str'),
            selection=dict(type='list', elements='str'),
            base_dn=dict(type='str')
        )),

        [...]

        users=dict(type='list', elements='dict', options=dict(
            source=dict(type='str'),
            selection=dict(type='list', elements='str'),
            base_dn=dict(type='str')

        [...]

        )),
    )
    argument_spec.update(checkpoint_argument_spec_for_objects)

    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
    api_call_object = 'access-role'

    result = api_call(module, api_call_object)
    module.exit_json(**result)
```



Thanks!
Thomas

**Comment by chkp-edenbr** [comment]
Indeed. A fix will be released in future release version

**Comment by chkp-edenbr** [comment]
Hi,
Fix now available in our new release v3.2.0.
Note that the "machines" and "users" arguments where split to two.
